src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -624,7 +624,7 @@ pub async fn compact(
                 );
             }
 
-            compaction_catalog_agent_ref
```

Comment:
Isn't `acquire_table_ids` always the same as `table_ids_to_be_compacted` if reached here? Filtering `pk_prefix_table_watermarks` and `non_pk_prefix_table_watermarks` via `table_ids_to_be_compacted` seems sufficient to me.

Reply:
Indeed, they are equal; otherwise, the compactor would abandon this task. Perhaps what you're wondering is why an extra pruning is needed. I just want to take a defensive measure. Functionally, the meta's pruning is sufficient.